packages/core/tsconfig.json

```diff
@@ -0,0 +1,12 @@
+{
+  "extends": "../../tsconfig.json",
+  "compilerOptions": {
+    "rootDir": "./src",
+    "outDir": "./dist/esm",
+    "declarationDir": "./dist/cjs"
```

Comment:
Shouldn't be this the same esm or cjs?

Reply:
Nope. This is only for .d.ts files. You reference them in the package.json file so it will point to the correct file. I think later down the line, we should add them to a separate folder, but it will need an additional script that will create a single .d.ts file that imports all partial module declarations.